### ISSUE TITLE
`test_run_with_user_global_config_file` fails if `XDG_CONFIG_HOME` is set

### ISSUE THREAD

**Issue by Jayman2000** [adrienverge/yamllint#621]
I’m running [NixOS](https://nixos.org), and on my system, the [`XDG_CONFIG_HOME`](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables) environment variable is set by default. This causes one of the unit tests to fail. You should be able to reproduce this issue on any system by running:

```bash
XDG_CONFIG_HOME="$PWD" python -m unittest tests.test_cli.CommandLineTestCase.test_run_with_user_global_config_file
```

I can workaround the problem by doing this:

```bash
env --unset=XDG_CONFIG_HOME python -m unittest tests.test_cli.CommandLineTestCase.test_run_with_user_global_config_file
```

Here’s what I think that the problem is: [The test](https://github.com/adrienverge/yamllint/blob/3288d0596e4522acfeb12bdfe8c1aaa3974e686c/tests/test_cli.py#L310) sets the `HOME` variable in an attempt to get yamllint to load an alternative global configuration file, but yamllint doesn’t look in the user’s home directory for a global configuration if `XDG_CONFIG_HOME` is set.

**Comment by adrienverge** [comment]
Hello Jason,

Thanks for reporting this, I can confirm the problem when `XDG_CONFIG_HOME` is set in environment.

> Here’s what I think that the problem is: [The test](https://github.com/adrienverge/yamllint/blob/3288d0596e4522acfeb12bdfe8c1aaa3974e686c/tests/test_cli.py#L310) sets the `HOME` variable in an attempt to get yamllint to load an alternative global configuration file, but yamllint doesn’t look in the user’s home directory for a global configuration if `XDG_CONFIG_HOME` is set.

I think you are right: https://github.com/adrienverge/yamllint/blob/3288d05/yamllint/cli.py#L184-L188.

Do you think deleting / restoring `XDG_CONFIG_HOME` in this problematic test would be a good solution? If yes, would you like to implement it?

**Comment by Jayman2000** [comment]
> Do you think deleting / restoring `XDG_CONFIG_HOME` in this problematic test would be a good solution?

Yeah.

> If yes, would you like to implement it?

Sure!
